packages/web-components/src/components/menu/menu-item.ts

```diff
@@ -0,0 +1,339 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2019, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { prefix } from '../../globals/settings';
+import styles from './menu-item.scss?lit';
+import { carbonElement as customElement } from '../../globals/decorators/carbon-element';
+import { classMap } from 'lit/directives/class-map.js';
+import CaretLeft16 from '@carbon/icons/lib/caret--left/16';
+import CaretRight16 from '@carbon/icons/lib/caret--right/16';
+import { consume } from '@lit/context';
+import { MenuContext } from './menu-context';
+import Checkmark16 from '@carbon/icons/lib/checkmark/16';
+
+/**
+ * Menu Item.
+ *
+ * @element cds-menu-item
+ */
+@customElement(`${prefix}-menu-item`)
+class CDSmenuItem extends LitElement {
+  @consume({ context: MenuContext })
+  context;
+  // @provide({ context: MenuContext })
+  // @consume({ context: MenuContext, subscribe: true })
+  // contextChild;
+
+  readonly hoverIntentDelay = 150; // in ms
+  hoverIntentTimeout;
+  /**
+   * Label for the menu item.
+   */
+  @property({ type: String })
+  label;
+  /**
+   * Children - Sub menu items.
+   */
+  @property({ type: Array })
+  childElements: Element[] | undefined;
+  /**
+   * Child context.
+   */
+  @property()
+  childContext;
+  /**
+   * Shortcut for the menu item.
+   */
+  @property({ type: String })
+  shortcut;
+  /**
+   * Sets the menu item's icon.
+   */
+  @property()
+  renderIcon?: () => void;
+  /**
+   * Disabled property for the menu item.
+   */
+  @property({ type: Boolean })
+  disabled;
+  /**
+   * Whether the menu item is disabled or not.
+   */
+  @property({ type: Boolean })
+  isDisabled;
+  /**
+   * Whether the menu submen for an item is open or not.
+   */
+  @property({ type: Boolean })
+  submenuOpen = false;
+  /**
+   * Whether the menu submen for an item is open or not.
+   */
+  @property({ type: Array })
+  submenuList;
+  /**
+   * Entrypoint.
+   */
+  @property()
+  submenuEntry;
+  /**
+   * Document direction.
+   */
+  @property({ type: String })
+  direction = 'ltr';
+  /**
+   * Checks if document direction is rtl.
+   */
+  @property({ type: Boolean })
+  isRtl = false;
+  /**
+   * Checks if document direction is rtl.
+   */
+  @property()
+  kind: 'default' | 'danger' = 'default';
```

Comment:
can we make a `defs` file for things like this? here's an example: https://github.com/carbon-design-system/carbon/blob/main/packages/web-components/src/components/ui-shell/defs.ts